coreneuron/io/nrn_setup.cpp

```diff
@@ -1026,47 +1034,53 @@ size_t model_size(void) {
         }
 
         // basic thread size includes mechanism data and G*V=I matrix
-        nb_nt += sz_th;
-        nb_nt += nt._ndata * szd + nt._nidata * szi + nt._nvdata * szv;
-        nb_nt += nt.end * szi;  // _v_parent_index
+        nb_nt += sz_nrnThread;
+        nb_nt += nt._ndata * sizeof(double) + nt._nidata * sizeof(int) + nt._nvdata * sizeof(void*);
+        nb_nt += nt.end * sizeof(int);  // _v_parent_index
+
+        // spike handling
+        nb_nt += nt.n_pntproc * sz_pntproc + nt.n_netcon * sz_netcon + nt.n_presyn * sz_presyn +
+                 nt.n_input_presyn * sz_input_presyn + nt.n_weight * sizeof(double);
+        nbyte += nb_nt;
 
 #ifdef DEBUG
         printf("ncell=%d end=%d nmech=%d\n", nt.ncell, nt.end, nmech);
         printf("ndata=%ld nidata=%ld nvdata=%ld\n", nt._ndata, nt._nidata, nt._nvdata);
         printf("nbyte so far %ld\n", nb_nt);
-        printf("n_presyn = %d sz=%ld nbyte=%ld\n", nt.n_presyn, sz_ps, nt.n_presyn * sz_ps);
-        printf("n_input_presyn = %d sz=%ld nbyte=%ld\n", nt.n_input_presyn, sz_psi,
-               nt.n_input_presyn * sz_psi);
-        printf("n_pntproc=%d sz=%ld nbyte=%ld\n", nt.n_pntproc, sz_pp, nt.n_pntproc * sz_pp);
-        printf("n_netcon=%d sz=%ld nbyte=%ld\n", nt.n_netcon, sz_nc, nt.n_netcon * sz_nc);
+        printf("n_presyn = %d sz=%ld nbyte=%ld\n", nt.n_presyn, sz_presyn, nt.n_presyn * sz_presyn);
+        printf("n_input_presyn = %d sz=%ld nbyte=%ld\n", nt.n_input_presyn, sz_input_presyn,
+               nt.n_input_presyn * sz_inputPreSyn);
+        printf("n_pntproc=%d sz=%ld nbyte=%ld\n", nt.n_pntproc, sz_pntproc, nt.n_pntproc * sz_pntproc);
+        printf("n_netcon=%d sz=%ld nbyte=%ld\n", nt.n_netcon, sz_netcon, nt.n_netcon * sz_netcon);
         printf("n_weight = %d\n", nt.n_weight);
```

Comment:
@ohm314 : just wondering : last week you added options for `--mech-stats`. What if we change that to `--model-stats` and also move this print memory information like this there? (we can print min / max / avg across ranks for different components)

Reply:
Yeah, i think it's not a bad idea.. I was actually consider the same (which is why i just pushed this as draft for now 😄 )

Reply:
Currently we have a lot of numbers printed in debug. Do we want to move all of those into model-stats? It seems maybe some are a bit redundant and overkill?